server/src/main/java/org/opensearch/action/admin/indices/close/TransportCloseIndexAction.java

```diff
@@ -172,4 +176,20 @@ protected void clusterManagerOperation(
             delegatedListener.onFailure(t);
         }));
     }
+
+    /**
+     * Reject close index request if cluster mode is [MIXED] and migration direction is [RemoteStore]
+     * @throws IllegalStateException if cluster mode is [MIXED] and migration direction is [RemoteStore]
+     */
+    private void validateRemoteMigration() {
+        ClusterSettings clusterSettings = clusterService.getClusterSettings();
+        CompatibilityMode compatibilityMode = clusterSettings.get(RemoteStoreNodeService.REMOTE_STORE_COMPATIBILITY_MODE_SETTING);
+        Direction migrationDirection = clusterSettings.get(RemoteStoreNodeService.MIGRATION_DIRECTION_SETTING);
+        if (compatibilityMode == CompatibilityMode.MIXED) {
+            boolean isRemoteMigration = migrationDirection == Direction.REMOTE_STORE;
```

Comment:
Can we merge the logic like this? Easier to read this way

```suggestion
            boolean isRemoteMigration = migrationDirection == Direction.REMOTE_STORE && compatibilityMode == CompatibilityMode.MIXED;
```